invertedai_drive/utils.py

```diff
@@ -81,24 +85,111 @@ def _request(
                 json=json,
             )
         except requests.exceptions.RequestException as e:
-            raise e
-        # TODO: Add logger
-        # TODO: Add low-level reponse parser, error handling
-        return result
+            raise error.APIConnectionError("Error communicating with IAI") from e
+        iai.logger.info(
+            iai.logger.logfmt(
+                "IAI API response",
+                path=self.base_url,
+                response_code=result.status_code,
+            )
+        )
+        data = self._interpret_response_line(result)
+        return data
 
     def _get_base_url(self) -> str:
         """
         This function returns the endpoint for API calls, which includes the
         version and other endpoint specifications.
         The method path should be appended to the base_url
         """
-        if not invertedai_drive.dev:
+        if not iai.dev:
             base_url = "https://api.inverted.ai/drive"
         else:
-            base_url = invertedai_drive.dev_url
+            base_url = iai.dev_url
         # TODO: Add endpoint option and versioning to base_url
         return base_url
 
+    def _handle_error_response(self, rbody, rcode, resp, rheaders, stream_error=False):
+        try:
+            error_data = resp["error"]
+        except (KeyError, TypeError):
+            raise error.APIError(
+                "Invalid response object from API: %r (HTTP response code "
+                "was %d)" % (rbody, rcode),
+                rbody,
+                rcode,
+                resp,
+            )
+
+        if "internal_message" in error_data:
```

Comment:
what is internal_message here ?

Reply:
This would be a place for us to send error data from the server, in case there is some discrepancy in the client's data.